### PR TITLE
[CSL-883] Tests/csl883 chrono

### PR DESCRIPTION
**Pull request by rockbmb**

### CODE REVIEW

File: test/Test/Pos/UtilSpec.hs [hunk 1]
```diff
@@ -112,3 +153,38 @@ verifyDiffMapIsSmaller (SmallHashMap hm1) (SmallHashMap hm2) =
...
+    :: (Arbitrary a, Show a, Eq a)
+    => a
+    -> Expectation
+toSingletonN x = one x `shouldBe` (NewestFirst @[] . one $ x)
+
+toSingletonO
+    :: (Arbitrary a, Show a, Eq a)
+    => a
+    -> Expectation
+toSingletonO x = one x `shouldBe` (OldestFirst @[] . one $ x)
```
Comment (volhovM) [inline test/Test/Pos/UtilSpec.hs]: I guess `toSingleton0` and `toSingletonN` can be merged and abstracted over constructors. I also don't like indentation.

File: test/Test/Pos/UtilSpec.hs [hunk 2]
```diff
@@ -112,3 +153,38 @@ verifyDiffMapIsSmaller (SmallHashMap hm1) (SmallHashMap hm2) =
...
+
+
+toFromList
+    :: forall f t a. (Arbitrary (t f a),
+                  Show (t f a),
+                  Eq (t f a),
+                  IL.IsList (f a),
+                  IL.IsList (t f a))
```
Comment (volhovM) [inline test/Test/Pos/UtilSpec.hs]: Can be fit into one line (nobody reads it carefully anyway).

File: test/Test/Pos/UtilSpec.hs (hunk 1 above)
Comment (Martoon-00) [inline test/Test/Pos/UtilSpec.hs]: Keh, I though indenting type like this is considered a cool style

File: test/Test/Pos/UtilSpec.hs (hunk 2 above)
Comment (Martoon-00) [inline test/Test/Pos/UtilSpec.hs]: And now you can type `Each [Arbitrary, Show, Eq, IL.IsList] '[t f a]`  🌟 
(`Each` is from universum. Note apostrophe before second list)

File: test/Test/Pos/UtilSpec.hs [hunk 3]
```diff
@@ -41,6 +66,22 @@ spec = describe "Util" $ do
         \ corresponding to these keys have a non-empty intersection, the difference\
         \ map's inner maps corresponding to those keys will be smaller in size than the\
         \ inner maps in the minuend hashmap"
+    description_One =
+        "Turning a single element into a chronological is the same as turning into the\
+        \ structure within and applying the outer data constructor."
+    description_toFromListNew functor =
+        "Converting 'NewestFirst " ++ functor ++ " a' to '[Item a]' and back changes\
```
Comment (Martoon-00) [inline test/Test/Pos/UtilSpec.hs]: Perhaps using `formatToString` by analogy to `sformat` will be more convinient here

File: test/Test/Pos/UtilSpec.hs (hunk 1 above)
Comment (volhovM) [inline test/Test/Pos/UtilSpec.hs]: Usually is when function is bigger than the type, yes. Hindent does this automatically btw.

**Comment by gromakovsky** [comment]
I think it makes sense two squash all four commits. Or at least the first two.

File: test/Test/Pos/UtilSpec.hs (hunk 2 above)
Comment (rockbmb) [inline test/Test/Pos/UtilSpec.hs]: @Martoon-00 Was this `Each` introduced with the recent changes to `universum`? My compiler's complaining it doesn't exist.

Comment (Martoon-00) [inline test/Test/Pos/UtilSpec.hs]: Yep, you'll have to use at least `universum-0.2.2` for it. You can just merge master as well

**Pull request merged**